src/frontend/screens/Settings/MapSettings/DeleteMapBottomSheet.tsx

```diff
@@ -0,0 +1,150 @@
+import * as React from "react";
+import BottomSheet, {
+  BottomSheetBackdrop,
+  BottomSheetView,
+} from "@gorhom/bottom-sheet";
+import { BottomSheetMethods } from "@gorhom/bottom-sheet/lib/typescript/types";
+import { StyleSheet, View, Text } from "react-native";
+import { MAPEO_BLUE, RED, WHITE } from "../../../lib/styles";
+import { DeleteIcon, ErrorIcon } from "../../../sharedComponents/icons";
+import Button from "../../../sharedComponents/Button";
+import { defineMessages, FormattedMessage, useIntl } from "react-intl";
+import api from "../../../api";
+import { useNavigationFromRoot } from "../../../hooks/useNavigationWithTypes";
+import { useMapStyle } from "../../../hooks/useMapStyle";
+import { DEFAULT_MAP_ID } from "./BackgroundMaps";
+
+const m = defineMessages({
+  deleteMapMessage: {
+    id: "screens.Settings.MapSettings.DeleteMapBottomSheet.deleteMapMessage",
+    defaultMessage: "Are you sure you want to delete {mapName}?",
+  },
+  deleteMapWarning: {
+    id: "screens.Settings.MapSettings.DeleteMapBottomSheet.deleteMapWarning",
+    defaultMessage:
+      "This area will no longer be available offline. Cannot be undone.",
+  },
+  deleteMap: {
+    id: "screens.Settings.MapSettings.DeleteMapBottomSheet.deleteMap",
+    defaultMessage: "Delete Map",
+  },
+  cancel: {
+    id: "screens.Settings.MapSettings.DeleteMapBottomSheet.cancel",
+    defaultMessage: "Cancel",
+  },
+});
+
+interface DeleteMapBottomSheetProps {
+  mapName: string;
+  mapId: string;
+  closeSheet: () => void;
+}
+
+export const DeleteMapBottomSheet = React.forwardRef<
+  BottomSheetMethods,
+  DeleteMapBottomSheetProps
+>(({ mapName, closeSheet, mapId }, sheetRef) => {
+  const [snapPoints, setSnapPoints] = React.useState<(number | string)[]>([
+    0,
+    "40%",
+  ]);
+
+  const { navigate } = useNavigationFromRoot();
+  const { formatMessage: t } = useIntl();
+  const { styleId, setStyleId } = useMapStyle();
+
+  function deleteMap() {
+    // Cannot delete Default Map
+    if (mapId === DEFAULT_MAP_ID) {
+      return;
+    }
+
+    api.maps
+      .deleteStyle(mapId)
+      .then(() => {
+        // If user is deleting the map that is currently being used, we want to set the map to be the default map
+        if (styleId === mapId) {
+          setStyleId(DEFAULT_MAP_ID);
+        }
+        navigate("BackgroundMaps");
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  }
+
+  return (
+    <BottomSheet
```

Comment:
not blocking but think we need to specify when to use bottom sheet vs bottom sheet modal. i think for the one spec'd in the design system, that will usually be a bottom sheet modal, whereas the one used in the main map screen should be a regular bottom sheet (as it's implemented right now). basically, the nature of the interaction for it will determine that (because the modal vs normal have different behaviors too), and maybe we need to consult Sabella on this too.

for where this component is used, I think this should technically be a modal too, but if it works okay then it's something we can revisit later.

started thinking about all of this stuff in https://github.com/digidem/mapeo-mobile/pull/1031 😅

Reply:
I agree, we should come up with some standard, it seems like even `@gorham/bottom-sheet` doesn't have a definition. I think generally modal implies some sort of navigation and/or blocking the rest of the UI, which is what is happening here, so Ill change it to a modal

Reply:
> I think generally modal implies some sort of navigation and/or blocking the rest of the UI

this is basically spot-on with what i was thinking. hopefully switching to our custom BottomSheetModal here will be less code too 😄